rules/downgrade-php74/src/Rector/FuncCall/DowngradeStripTagsCallWithArrayRector.php

```diff
@@ -0,0 +1,184 @@
+        } elseif ($allowableTagsParam instanceof Variable || $allowableTagsParam instanceof PropertyFetch || $allowableTagsParam instanceof ConstFetch || $allowableTagsParam instanceof ClassConstFetch) {
+            // If it is a variable or a const (other than null), add logic to maybe convert to string
+            $newExpr = $this->getIfArrayConvertArrayToStringFuncCall($allowableTagsParam);
+        } else {
+            // It is a function or method call, ternary or coalesce, or any other:
+            // Assign the value to a variable
+            // First obtain a variable name that does not exist in the node (to not override its value)
+            $variableName = $this->variableNaming->resolveFromFuncCallFirstArgumentWithSuffix(
+                $node,
+                'AllowableTags',
+                'allowableTags',
+                $node->getAttribute(AttributeKey::SCOPE)
+            );
+            // Assign the value to the variable
+            $newVariable = new Variable($variableName);
+            $this->addNodeBeforeNode(new Assign($newVariable, $allowableTagsParam), $node);
+
+            // Apply refactor on the variable
+            $newExpr = $this->getIfArrayConvertArrayToStringFuncCall($newVariable);
+        }
+
+        // Replace the arg with a new one
+        array_splice($node->args, 1, 1, [new Arg($newExpr)]);
+        return $node;
+    }
+
+    /**
+     * @param FuncCall $node
+     */
+    private function shouldRefactor(Node $node): bool
+    {
+        if (! $this->isFuncCallName($node, 'strip_tags')) {
+            return false;
+        }
+
+        // If param not provided, do nothing
+        if (count($node->args) < 2) {
+            return false;
+        }
+
+        // Process anything other than String and null (eg: variables, function calls)
+        $allowableTagsParam = $node->args[1]->value;
+
+        // Skip for string
+        if ($allowableTagsParam instanceof String_) {
+            return false;
+        }
+        // Skip for null
+        // Allow for everything else (Array_, Variable, PropertyFetch, ConstFetch, ClassConstFetch, FuncCall, MethodCall, Coalesce, Ternary, others?)
+        return ! $this->isNull($allowableTagsParam);
+    }
+
+    private function getConvertArrayToStringFuncCall(Expr $expr): Expr
+    {
+        return new Concat(
+            new Concat(
+                new String_('<'),
+                new FuncCall(new Name('implode'), [new Arg(new String_('><')), new Arg($expr)])
+            ),
+            new String_('>')
+        );
+    }
+
+    private function getIfArrayConvertArrayToStringFuncCall(Expr $expr): Expr
+    {
+        return new Ternary(
+            new BooleanAnd(
+                new NotIdentical($expr, $this->createNull()),
+                new FuncCall(new Name('is_array'), [new Arg($expr)])
+            ),
+            $this->getConvertArrayToStringFuncCall($expr),
+            $expr
+        );
+    }
```

Comment:
These should be named like `createXFromY()` with more narrow param/return types

```diff
-private function getIfArrayConvertArrayToStringFuncCall(Expr $expr): Expr
+private function createFuncCallFromArray(Array_ $array): FuncCall
```

- It helps with better static control
- The `create*()` method name makes clear the node is completely new, not just modified

Reply:
Ok, please check if the new function names are OK. I can't narrow the param type from `Expr` since it can receive many types, so I made them explicit in the phpdoc

Reply:
Looks good :+1: